tools/testing/test_selections.py

```diff
@@ -30,43 +32,75 @@
         NUM_PROCS = 1
 
 
+class ShardedTest(NamedTuple):
+    name: str
+    shard: int
+    num_shards: int
+    time: Optional[float]
+
+    def __str__(self) -> str:
+        if self.num_shards == 1:
+            return self.name
```

Comment:
Do you think having f"{self.name} {self.shard}/{self.num_shards}" for all cases would make it easier to parse the name later on? i.e. "my_test 1/1" is ok I think